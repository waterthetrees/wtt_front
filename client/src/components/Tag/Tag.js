import React from 'react';
import './Tag.scss';

export const Tag = ({ variant, children }) => {
  const tagVariant = `tag__${variant}`;
  return (
    <div className="tag">
      <div className={tagVariant}>{children}</div>
    </div>
  );
};
