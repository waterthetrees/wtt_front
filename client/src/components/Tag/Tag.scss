.tag {
  margin: 0.5rem;
  color: white;
  font-size: small;
  text-decoration: none;
  transition: all 0.2s ease-in-out;
  color: white;
  width: fit-content;

  &__brown {
    background-color: #795548;
    border-radius: 15px;
    padding: 5px 10px;
  }
  &__green {
    background-color: green;
    border-radius: 15px;
    padding: 5px 10px;
  }
}
