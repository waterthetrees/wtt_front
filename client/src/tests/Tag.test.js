import React from 'react';
import { render } from '@testing-library/react';
import { Tag } from '@/components/Tag/Tag';

describe('<Tag /> spec', () => {
  it('Renders the Tag component correctly', () => {
    const { container } = render(<Tag variant="green">Test Tag</Tag>);
    expect(container).toMatchSnapshot();
  });

  it('Renders children inside the Tag component', () => {
    const { getByText } = render(<Tag variant="green">Test Tag</Tag>);
    const tagElement = getByText('Test Tag');
    expect(tagElement).not.toBeNull();
  });

  it('Applies the correct variant class to the inner div', () => {
    const { getByText } = render(<Tag variant="brown">Test Tag</Tag>);
    const innerDiv = getByText('Test Tag');
    expect(innerDiv.className).toContain('tag__brown');
  });
});
