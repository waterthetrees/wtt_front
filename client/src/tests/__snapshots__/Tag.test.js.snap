// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag /> spec Renders the Tag component correctly 1`] = `
<div>
  <div
    class="tag"
  >
    <div
      class="tag__pill tag__green"
    >
      Test Tag
    </div>
  </div>
</div>
`;
